openhands_ext/test_storage.py

```
@@ -0,0 +1,207 @@
+    In enterprise, this would be a database with proper encryption
+    """
+    
+    def __init__(self, base_path: str = os.getenv("TEST_STORAGE_PATH", "/tmp/openhands-test-storage")):
+        self.base_path = Path(base_path)
+        self.base_path.mkdir(exist_ok=True)
+    
+    def _get_user_path(self, user_id: str) -> Path:
+        """Get user-specific storage path (tenant isolation)"""
+        # Sanitize user_id to prevent path traversal attacks
+        if '..' in user_id or os.path.sep in user_id or not re.match(r'^[a-zA-Z0-9._-]+$', user_id):
+            raise ValueError("Invalid user_id, path traversal attempt detected.")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)